eng/tsp-core/pipelines/jobs/build-for-publish.yml

```diff
@@ -10,6 +10,9 @@ jobs:
         parameters:
           useDotNet: true
 
+      - script: pnpm run update-telemetry-key $(vscode-telemetry-key)
+        workingDirectory: $(Build.SourcesDirectory)/packages/typespec-vscode
+
```

Comment:
@timotheeguerin , I need to update the telemetry-key to the PROD one when publishing vscode extension. Do you know how I can add the PROD key into the $(vscode-telemetry-key) varaible? And how can I test whether the updated pipeline works without actually doing a publish? thanks.

Reply:
no, it doesn't need to be private.

Reply:
Talked with guys from AzureApiCenter vscode extension team. Although it's not sensitive and will be in the released package.json eventually, but they still store the key as secret var in pipeline instead of plain text in github (https://github.com/microsoft/vscode-azureapicenter/blob/dev/.github/workflows/cd.yml#L39-L44). I think we can just follow it and I have tried added a secret to the "typespec -Publish" pipeline for it (https://dev.azure.com/azure-sdk/internal/_apps/hub/ms.vss-build-web.ci-designer-hub?pipelineId=3226&nonce=wqPmyYzOeUiSS5fefsbaeA%3D%3D&branch=main), but do you know how I can test the change without actually publish something?

Reply:
nvm. Verified by comment the publish part. thanks.

Reply:
it does feel a little inconsistent that we do that for this key but not the website

Reply:
Not quite sure about the Key in website. But in general, I would suggest not to check in these kinds of stuff in code directly especially when it's very easy to maintain them in pipeline. I can help to make the script more general so that it can be reused by website if needed. Just let me know. thanks.